code/_onclick/xeno.dm

```diff
@@ -37,12 +37,34 @@
 		if(XENO_NO_DELAY_ACTION)
 			next_move = world.time
 		else
-			if(!tile_attack)
+			if(!tile_attack) // Patting flames for Xenos
+				var/firepatted = FALSE
+				if(src.a_intent == INTENT_HELP)
+					var/fire_level_to_extinguish = 5
+					var/turf/T = target
```

Comment:
Not sure if target is guaranteed to be a turf at this point. Also T should be named better.

Reply:
It has to be because we're in the isturf(target) here.